File: 1000-1999/1295. Find Numbers with Even Number of Digits/Solution.cpp
/**
 * Author: Fbin
 * Link:https://leetcode.com/problems/find-numbers-with-even-number-of-digits/description/?envType=daily-question&envId=2025-04-30
 * Created at: 30/04/2025
 */

#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int findNumbers(vector<int>& nums) {
    int ans = 0;
    for (int i = 0; i < nums.size(); i++) {
      int count = 0;
      while (nums[i] > 0) {
        nums[i] = nums[i] / 10;
        count++;
      }
      ans += 1 - count % 2;
    }
    return ans;
  }
};

File: 1000-1999/1295. Find Numbers with Even Number of Digits/Solution.ts
function findNumbers(nums: number[]): number {
	let ans = 0;
	for (let i = 0; i < nums.length; i++) {
		let count = 0;
		while (nums[i] > 0) {
			nums[i] = Math.floor(nums[i] / 10);
			count++;
		}
		if(count % 2 == 0) ans++;
	}

	return ans;
}
